drivers/regulator/regulator_da1469x.c

```diff
@@ -281,6 +286,20 @@ static int regulator_da1469x_set_voltage(const struct device *dev, int32_t min_u
 	uint16_t idx;
 	uint32_t mask;
 
+	if ((SystemCoreClock == PLL_FREQ) && (dev == VDD_DEVICE)) {
```

Comment:
Can you just check the enum here?

```suggestion
	if ((SystemCoreClock == PLL_FREQ) && (config->rail == VDD)) {
```

Reply:
Nice catch. It turned out that `rail` property was never initialized and thus it has zero value. I applied your suggestion.